File: Application/src/main/java/com/shareyourproxy/api/rx/command/eventcallback/UserContactDeletedEventCallback.java
package com.shareyourproxy.api.rx.command.eventcallback;

import android.os.Parcel;

import com.shareyourproxy.api.domain.model.User;

/**
 * Created by Evan on 6/8/15.
 */
public class UserContactDeletedEventCallback extends UserEventCallback {
    private final static java.lang.ClassLoader CL =
        UserContactDeletedEventCallback.class.getClassLoader();
    public static final Creator<UserContactDeletedEventCallback> CREATOR =
        new Creator<UserContactDeletedEventCallback>() {
            @Override
            public UserContactDeletedEventCallback createFromParcel(Parcel in) {
                return new UserContactDeletedEventCallback(
                    (User) in.readValue(CL), (String) in.readValue(CL));
            }

            @Override
            public UserContactDeletedEventCallback[] newArray(int size) {
                return new UserContactDeletedEventCallback[size];
            }
        };
    private final String contactId;

    public UserContactDeletedEventCallback(User user, String contactId) {
        super(user);
        this.contactId = contactId;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(contactId);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/UserContactDeletedEventCallback.kt
package com.shareyourproxy.api.rx.command.eventcallback

import android.os.Parcel
import android.os.Parcelable

import com.shareyourproxy.api.domain.model.User

/**
 * User contact has been deleted.
 */
class UserContactDeletedEventCallback(user: User, private val contactId: String) : UserEventCallback(user) {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(contactId)
    }

    companion object {
        private val CL = UserContactDeletedEventCallback::class.java.classLoader
        val CREATOR: Parcelable.Creator<UserContactDeletedEventCallback> = object : Parcelable.Creator<UserContactDeletedEventCallback> {
            override fun createFromParcel(parcel: Parcel): UserContactDeletedEventCallback {
                return UserContactDeletedEventCallback(
                        parcel.readValue(CL) as User, parcel.readValue(CL) as String)
            }

            override fun newArray(size: Int): Array<UserContactDeletedEventCallback?> {
                return arrayOfNulls(size)
            }
        }
    }
}
